### DISCUSSION TITLE
Discussion: How to analyze flattened Policy Engine file versus taxdata

### DISCUSSION THREAD

**Discussion by donboyd5** [PSLmodels/tax-microdata-benchmarking#217] [Q&A]
@nikhilwoodruff, @martinholmer 

I've started a preliminary ad hoc analysis of the Feb 14 version of the flattened PolicyEngine file, comparing it to the taxdata file grown to 2023, provided by @martinholmer , and to a version of that taxdata file with only the variables that the flattened PolicyEngine file has. The results so far are at [this web page](https://boyd-psl-adhoc.netlify.app/). I do not pretend that the taxdata file is truth. Still, there are some large differences that eventually would be important to understand.

While I don't think there is anything we have to do with the results in Phase 1, there are lessons and perhaps some pointers that might be useful even in Phase 1. I think we should discuss briefly in our next call.

I'll plan to update this with some more tables as time permits, and will rerun when there is a new version of the flattened PolicyEngine file. I do not plan to save analyses of older versions of the file unless you tell me you think it is important.

**Comment by nikhilwoodruff** [comment]
Thanks Don- this is really useful! Is the code for this available so I can re-run it more frequently as the data updates?

**Comment by donboyd5** [comment]
I'll get it to github later today. It's R code and uses quarto, so you'd
need to install quarto on your computer. That's easy.

Don

**Comment by nikhilwoodruff** [comment]
OK thanks. Yep, just installed.

**Comment by donboyd5** [comment]
The repo is [here](https://github.com/donboyd5/taxdata-psl). There is some deadwood in it (copied over from other projects) and it loads far more packages than it actually uses right now (copied over from other projects). 

The quarto project that creates the ad hoc web page is in the folder pe_flatfile.

You'll see that it crudely calls tc.exe from R, in Windows. You'll need to adjust that to work on your system.

There is some code you'll have to run onetime - e.g., creating an rds file with tax-calculator documentation for variables - and then you can set eval false for the relevant code chunk.

The _quarto.yml file directs traffic so you should start there. All sorts of stuff commented out to remind me how to do things as I extend the project. Once you have everything set up properly, you should be able to run everything by opening a terminal in pe_flatfile folder and doing "quarto render" or "quarto publish netlify --no-browser --no-prompt" if you have a netlify account.

I put it all together quickly with me as the audience so it is not as clear as it might be. If you have trouble getting it to work, pls let me know.

Don

**Comment by donboyd5** [comment]
The R project uses my package [btools](https://github.com/donboyd5/btools). You can install it or just ignore it. 

I think the only things the ad hoc project uses are:

- the function `ht()`, which shows the head and tail of a data frame or other object; obviously, there are plenty of workarounds for that
- the function `ns()`, which shows the names of an object sorted in alpha order; again, easy workarounds

**Comment by donboyd5** [comment]
@nikhilwoodruff @martinholmer 

I have updated the code to download the latest PE flat file and have run the code on that file. Results are in the [same place as before](https://boyd-psl-adhoc.netlify.app/), overwriting the results for the older PE file.

The code is on GitHub and is on the web page as well, folded.

**Comment by donboyd5** [comment]
Fixed code error as discussed in this [PR](https://github.com/PSLmodels/tax-microdata-benchmarking/pull/17#pullrequestreview-1903683311).
